### PR TITLE
Support of implicit foreign key of type atlas.rpc.Identifier.

### PR DESCRIPTION
**Pull request by amaskalenka**
The intention of added `reference_of` option is to allow calculate resource type of `atlas.rpc.Identifier` properly in case when an identifier is implicit foreign key.

NOTE: The **implicit foreign key** is a foreign key defined on DB level via constraints but in Proto file it is defined as simple field that holds id of some object without any association.

### CODE REVIEW

**Comment by Evgeniy-L** [comment]
looks ok to me

File: example/user/user.proto [hunk 1]
```diff
@@ -55,6 +55,7 @@ message Address {
     string address_2 = 3;
     string post = 4;
     atlas.rpc.Identifier external = 5 [(gorm.field).tag = {type: "jsonb"}];
+    atlas.rpc.Identifier implicit_fk = 6 [(gorm.field) = {reference_of: "Email" tag: {type: "text"} }];
```
Comment (Calebjh) [inline example/user/user.proto]: Is there not a comma needed between `"Email" tag:`?

Comment (amaskalenka) [reply example/user/user.proto]: The comma is not required

**Comment by amaskalenka** [comment]
Exactly! That is why I call it implicit foreign key.

**Pull request merged**